### DISCUSSION TITLE
mTerm in MPC-Objective Function

### DISCUSSION THREAD

**Discussion by 19MTG04** [do-mpc/do-mpc#323] [Q&A] [answered]
Hello,
I´m trying to set the mterm in my MPC-Objective Function for an algebraic varaiable (z). The documentation shows that this is not possible at the moment.
Is there a specific reason why (integration issues or so) or could it be possible to use algebraic variables in the mterm in future releases?

Thanks

**Comment by 4flixt** [comment, marked as answer]
Hi @19MTG04,

thank you for your question. Algebraic variables are not well defined for the last step of the MPC horizon and therefore not available for the mterm. This is not a shortcoming of do-mpc and will not be changed in the future.

Best,
4flixt

**Reply by jrbull** [reply]
If I can't use algebraic variables, can auxiliary expressions be used to define the objective function?
do-mpc is perfect for my needs in all ways, except for this issue, which is a game stopper for me. Can you suggest a workaround? I could increase the DAE index by differentiating some z variables in time, but that increases the complexity and presumably solution time.

**Reply by jrbull** [reply]
Scratch that, I just found out that algebraic variables are allowed in the stage cost, lterm - that should be sufficient for my needs.
